src/js/background.js

```diff
@@ -0,0 +1,214 @@
+        `https://api.chess.com/pub/player/${playerName}/games/archives`,
+        tabId
+      )
+    ).archives;
+    /*
+     * Archives are given from the oldest to the most recet games.
+     * Thus iterating from the last one (the most recent) is usually
+     * better, because players usually want to analyse their most recent
+     * games. You will probably make less fetches
+     */
+    for (let i = archives.length - 1; i >= 0; i--) {
+      const games = (await fetchJSON(archives[i], tabId)).games;
+      for (let j = games.length - 1; j >= 0; j--) {
+        if (games[j].url === gameUrl) {
+          return games[j].pgn;
+        }
+      }
+    }
+  } catch (error) {
+    sendLogMessage(error, tabId);
+    return null;
+  }
+
+  return null;
+};
+
+const getPGNManual = async (tabId) => {
+  await waitAndClick(".share");
+  await waitAndClick(".board-tab-item-underlined-component");
+  await waitAndClick(".share-menu-tab-pgn-toggle");
+
+  if (await waitForElement("[name='pgn']")) {
+    const [pgn] = await browser.tabs.executeScript(tabId, {
+      code: `document.querySelector("[name='pgn']").value;`,
+    });
+    await waitAndClick("[data-cy='share-menu-close']");
+
+    return pgn;
+  }
+
+  return null;
+};
+
+const clickElement = async (querySelector, tabId) => {
+  await browser.tabs.executeScript(tabId, {
+    code: `document.querySelector("${querySelector}").click();`,
+  });
+};
+
+const waitForElement = async (
+  querySelector,
+  tabId,
+  timeLeft = 5000,
+  retryDelay = 100
+) => {
+  const elementFoundScript = `document.querySelector("${querySelector}") !== null;`;
+
+  let isElement = null;
+
+  while (!isElement && timeLeft > 0) {
+    await new Promise((r) => setTimeout(r, retryDelay));
+    timeLeft -= retryDelay;
+    [isElement] = await browser.tabs.executeScript(tabId, {
+      code: elementFoundScript,
+    });
+  }
+
+  if (!isElement)
+    sendLogMessage(`Cannot find \`${querySelector}\` element!`, tabId);
+
+  return isElement;
+};
+
+const waitAndClick = async (querySelector, tabId) => {
+  if (await waitForElement(querySelector, tabId))
+    await clickElement(querySelector, tabId);
+};
+
+const lichessAnalyse = async (tabId, pgn) => {
+  if (await waitForElement("[name='analyse']", tabId)) {
+    const [loggedIn] = await browser.tabs.executeScript(tabId, {
+      code: `!document.querySelector("[name='analyse']").disabled;`,
+    });
+    if (loggedIn) await waitAndClick("[name='analyse']", tabId);
+
+    if (await waitForElement("[name='pgn']", tabId)) {
+      await browser.tabs.executeScript(tabId, {
+        code: `document.querySelector("[name='pgn']").value = \`${pgn}\`;`,
+      });
+      await waitAndClick(".submit", tabId);
+
+      /*
+       * Bugfix where status was loading and firefox throwed
+       * an error about invalid host permissions
+       */
+      let status = "loading";
+      while (status === "loading") {
+        let getting = await browser.tabs.get(tabId);
```

Comment:
Could be `const`, I believe. Also, the variable name could be something more meaningful :smile: 

```suggestion
        const tab = await browser.tabs.get(tabId);
```